XmlFormatter/src/Windows/HotfolderEditor.cs

```diff
@@ -0,0 +1,227 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Windows.Forms;
+using XmlFormatter.src.Enums;
+using XmlFormatter.src.Hotfolder;
+using XmlFormatter.src.Interfaces.Formatter;
+using XmlFormatter.src.Interfaces.Hotfolder;
+
+namespace XmlFormatter.src.Windows
+{
+    /// <summary>
+    /// This Window is used to create or edit a hotfolder configuration
+    /// </summary>
+    public partial class HotfolderEditor : Form
+    {
+        /// <summary>
+        /// Is the editor in the hotfolder edit mode
+        /// </summary>
+        private readonly bool editMode;
+
+        /// <summary>
+        /// Was something saved
+        /// </summary>
+        private bool saved;
+
+        /// <summary>
+        /// Readonly access if something was saved
+        /// </summary>
+        public bool Saved => saved;
+
+        /// <summary>
+        /// A lookup for the formatters
+        /// </summary>
+        private Dictionary<string, Type> formatters;
+
+        /// <summary>
+        /// The current hotfolder configuration
+        /// </summary>
+        private IHotfolder hotfolder;
+
+        /// <summary>
+        /// Readonly access to the hotfolder configuration
+        /// </summary>
+        public IHotfolder Hotfolder => hotfolder;
+
+        /// <summary>
+        /// Create a new instance of the editor
+        /// </summary>
+        /// <param name="hotfolder">The hotfolder configuration to use for editing</param>
+        public HotfolderEditor(IHotfolder hotfolder)
+        {
+            editMode = hotfolder == null ? false : true;
+            formatters = new Dictionary<string, Type>();
+           
+            InitializeComponent();
+
+            CB_Formatter.Enabled = !editMode;
+            this.hotfolder = hotfolder;
+        }
+
+
+        /// <summary>
+        /// Loading the window
+        /// </summary>
+        /// <param name="sender">Sender of the event</param>
+        /// <param name="e">Event arguments</param>
+        private void HotfolderEditor_Load(object sender, EventArgs e)
+        {
+            Type type = typeof(IFormatter);
+            Assembly[] assemblies = AppDomain.CurrentDomain.GetAssemblies();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'assemblies'.](https://app.codacy.com/manual/simonaberle/XmlFormatter/pullRequest?prid=5053278)